commit 687c06394b5da617ef3e9c9c0482217cfb0881b0
Author: liangjianqiu <liangjianqiu@hkzechin.com>
Date:   Sun Oct 9 16:36:56 2016 +0800

    [Bmobile][Browser]根据MCC/MNC显示不同国家主页

diff --git a/alps/packages/apps/Browser/res/values/strings.xml b/alps/packages/apps/Browser/res/values/strings.xml
index afef421..9992bb6 100755
--- a/alps/packages/apps/Browser/res/values/strings.xml
+++ b/alps/packages/apps/Browser/res/values/strings.xml
@@ -798,7 +798,22 @@
     <!-- The default homepage. -->
     <!--<string name="homepage_base" translatable="false">
         http://www.google.com/webhp?client={CID}&amp;source=android-home</string>-->
-    <string name="homepage_base" translatable="false">http://www.google.com</string>
+   <string name="homepage_base" translatable="false">http://www.google.com</string>
+   <string name="homepage_base_argentina" translatable="false">http://wap.movistar.com.ar</string>
+   <string name="homepage_base_chile" translatable="false">wap.movistar.cl</string>
+   <string name="homepage_base_colombia" translatable="false">http://www.clarosva.com.co</string>
+   <string name="homepage_base_costa_rica" translatable="false">http://www.movistar.co.cr</string>
+   <string name="homepage_base_ecuador" translatable="false">http://www.movistar.com.ec</string>
+   <string name="homepage_base_el_salvador" translatable="false">http://www.movistar.com.sv</string>
+   <string name="homepage_base_guatemala" translatable="false">http://www.movistar.com.gt</string>
+   <string name="homepage_base_mexico" translatable="false">http://wap.movistar.com.mx</string>
+   <string name="homepage_base_nicaragua" translatable="false">http://www.movistar.com.ni</string>
+   <string name="homepage_base_panama" translatable="false">http://www.movistar.com.pa</string>
+   <string name="homepage_base_peru" translatable="false">http://www.movistar.com.pe</string>
+   <string name="homepage_base_uruguay" translatable="false">wap.movistar.com.uy</string>
+   <string name="homepage_base_venezuela" translatable="false">http://www.movistar.com.ve</string>
+   <string name="homepage_base_espana" translatable="false">http://emocion.terra.es</string>
+   
 
     <!-- Bookmarks -->
     <string-array name="bookmarks" translatable="false">
diff --git a/alps/packages/apps/Browser/src/com/android/browser/BrowserSettings.java b/alps/packages/apps/Browser/src/com/android/browser/BrowserSettings.java
index 4c9ab42..dbc2ac9 100755
--- a/alps/packages/apps/Browser/src/com/android/browser/BrowserSettings.java
+++ b/alps/packages/apps/Browser/src/com/android/browser/BrowserSettings.java
@@ -69,7 +69,7 @@ import java.util.Iterator;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.WeakHashMap;
-
+import android.telephony.TelephonyManager;
 /**
  * Class for managing settings
  */
@@ -277,7 +277,8 @@ public class BrowserSettings implements OnSharedPreferenceChangeListener,
                 /*if (BrowserFeatureOption.BROWSER_SITE_NAVIGATION_SUPPORT) {
                     sFactoryResetUrl = mContext.getResources().getString(R.string.homepage_base_site_navigation);
                 } else {*///lijun modify to set homepage as default
-                    sFactoryResetUrl = mContext.getResources().getString(R.string.homepage_base);
+                    //sFactoryResetUrl = mContext.getResources().getString(R.string.homepage_base);
+                    sFactoryResetUrl = setOtherhomepage(mContext);
                     if (sFactoryResetUrl.indexOf("{CID}") != -1) {
                         sFactoryResetUrl = sFactoryResetUrl.replace("{CID}",
                                 BrowserProvider.getClientId(mContext.getContentResolver()));
@@ -294,13 +295,106 @@ public class BrowserSettings implements OnSharedPreferenceChangeListener,
         }
     };
 
+
+    public static String setOtherhomepagestatic(Context context){
+
+        	TelephonyManager mGeminiPhone = (TelephonyManager)context.getSystemService("phone");
+
+              if(mGeminiPhone != null){
+				 String mGeminiSim1 = mGeminiPhone.getSimOperatorGemini(0);
+
+				 //Log.d(TAG, "mGeminiSim1 == " + mGeminiSim1);
+				 
+			 if(mGeminiSim1 != null &&(mGeminiSim1.equals("72207") ||mGeminiSim1.equals("46001"))){
+			 	return context.getResources().getString(R.string.homepage_base_argentina);		 	
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("73002")){
+			       return context.getResources().getString(R.string.homepage_base_chile);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("73002")){
+			       return context.getResources().getString(R.string.homepage_base_chile);
+			 }else if(mGeminiSim1 != null && (mGeminiSim1.equals("732123") || mGeminiSim1.equals("732101"))){
+			       return context.getResources().getString(R.string.homepage_base_colombia);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("71204")){
+			       return context.getResources().getString(R.string.homepage_base_costa_rica);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("74000")){
+			       return context.getResources().getString(R.string.homepage_base_ecuador);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("70604")){
+			       return context.getResources().getString(R.string.homepage_base_el_salvador);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("70403")){
+			       return context.getResources().getString(R.string.homepage_base_guatemala);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("33403")){
+			       return context.getResources().getString(R.string.homepage_base_mexico);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("71030")){
+			       return context.getResources().getString(R.string.homepage_base_nicaragua);
+			 }else if(mGeminiSim1 != null && (mGeminiSim1.equals("71402") || mGeminiSim1.equals("714020"))){
+			       return context.getResources().getString(R.string.homepage_base_panama);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("71606")){
+			       return context.getResources().getString(R.string.homepage_base_peru);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("74807")){
+			       return context.getResources().getString(R.string.homepage_base_uruguay);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("73404")){
+			       return context.getResources().getString(R.string.homepage_base_venezuela);
+			 }else if(mGeminiSim1 != null && (mGeminiSim1.equals("21402") || mGeminiSim1.equals("21407"))){
+			       return context.getResources().getString(R.string.homepage_base_espana);
+			 }
+		 
+             }	
+		return context.getResources().getString(R.string.homepage_base);
+    	}
+
+    public  String setOtherhomepage(Context context){
+
+        	TelephonyManager mGeminiPhone = (TelephonyManager)context.getSystemService("phone");
+
+              if(mGeminiPhone != null){
+				 String mGeminiSim1 = mGeminiPhone.getSimOperatorGemini(0);
+
+				 //Log.d(TAG, "mGeminiSim1 == " + mGeminiSim1);
+				 
+			 if(mGeminiSim1 != null &&(mGeminiSim1.equals("72207") ||mGeminiSim1.equals("46001"))){
+			 	return context.getResources().getString(R.string.homepage_base_argentina);		 	
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("73002")){
+			       return context.getResources().getString(R.string.homepage_base_chile);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("73002")){
+			       return context.getResources().getString(R.string.homepage_base_chile);
+			 }else if(mGeminiSim1 != null && (mGeminiSim1.equals("732123") || mGeminiSim1.equals("732101"))){
+			       return context.getResources().getString(R.string.homepage_base_colombia);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("71204")){
+			       return context.getResources().getString(R.string.homepage_base_costa_rica);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("74000")){
+			       return context.getResources().getString(R.string.homepage_base_ecuador);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("70604")){
+			       return context.getResources().getString(R.string.homepage_base_el_salvador);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("70403")){
+			       return context.getResources().getString(R.string.homepage_base_guatemala);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("33403")){
+			       return context.getResources().getString(R.string.homepage_base_mexico);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("71030")){
+			       return context.getResources().getString(R.string.homepage_base_nicaragua);
+			 }else if(mGeminiSim1 != null && (mGeminiSim1.equals("71402") || mGeminiSim1.equals("714020"))){
+			       return context.getResources().getString(R.string.homepage_base_panama);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("71606")){
+			       return context.getResources().getString(R.string.homepage_base_peru);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("74807")){
+			       return context.getResources().getString(R.string.homepage_base_uruguay);
+			 }else if(mGeminiSim1 != null && mGeminiSim1.equals("73404")){
+			       return context.getResources().getString(R.string.homepage_base_venezuela);
+			 }else if(mGeminiSim1 != null && (mGeminiSim1.equals("21402") || mGeminiSim1.equals("21407"))){
+			       return context.getResources().getString(R.string.homepage_base_espana);
+			 }
+		 
+             }	
+		return context.getResources().getString(R.string.homepage_base);
+    	}	
+
+
     public static String getFactoryResetUrlFromRes(Context context) {
         /// M: Operator Feature get the customer homepage. @{
         sBrowserSmallFeatureEx = Extensions.getSmallFeaturePlugin(context);
         sFactoryResetUrl = sBrowserSmallFeatureEx.getCustomerHomepage();
         
         if (sFactoryResetUrl == null) {
-            sFactoryResetUrl = context.getResources().getString(R.string.homepage_base);
+            //sFactoryResetUrl = context.getResources().getString(R.string.homepage_base);
+            sFactoryResetUrl = setOtherhomepagestatic(context);
         }
 
         if (sFactoryResetUrl.indexOf("{CID}") != -1) {
